src/OpenTelemetry.Instrumentation.AWS/Implementation/SqsRequestContextHelper.cs

```diff
@@ -38,23 +36,9 @@ internal static void AddAttributes(IRequestContext context, IReadOnlyDictionary<
             return;
         }
 
-        int nextAttributeIndex = attributesCount + 1;
         foreach (var param in attributes)
         {
-            AddAttribute(parameters, originalRequest, param.Key, param.Value, nextAttributeIndex);
-            nextAttributeIndex++;
+            originalRequest.MessageAttributes.Add(param.Key, new MessageAttributeValue { DataType = "String", StringValue = param.Value });
```

Comment:
This is altering the users provided object without them knowing it. If they are reusing request objects this will cause unintended side effects.

Reply:
Agree about potential side effects - but this is existing behavior for this library. 

Recommend leaving as is for this PR